src/plotting.jl

```diff
@@ -15,6 +15,28 @@ function heatmap_basins_attractors end
 function heatmap_basins_attractors! end
 export heatmap_basins_attractors, heatmap_basins_attractors!
 
+""" 
+    shaded_basins_heatmap(grid, basins, iterations, attractors; kwargs...) 
+
```

Comment:
This is a specialized application. It really only makes sense for 2D systems (or systems projected into 2D). 

That is why I think you should take the following code:
```
# basins, attractors = basins_of_attraction(mapper; show_progress = false)
basins = zeros(Int8, 1500, 1500); iterations = zeros(Int32,1500,1500)
for (i,x) in enumerate(xg), (j,y) in enumerate(yg) 
    l = mapper([x,y])
    n = get_iterations(mapper)
    basins[i,j] = l
    iterations[i,j] = n
end
using StatsBase:median
attractors = extract_attractors(mapper)
```

and make it into a function `iterates_of_basins_of_attraction` that takes the mapper and grid and returns `basins, attractors, iterations`. This way you can give the output of the fnction directly into the plotted function.